src/components/tables/hooks.ts

```diff
@@ -0,0 +1,28 @@
+import { useState } from 'react';
+import { SortDirection } from 'types';
+import { getPagination } from './EntityTable';
+
+function useTableState(defaultSortCol: any, defaultSortDir?: SortDirection) {
```

Comment:
Taken from the repeated stuff in table components. More should go in here eventually... maybe. Some of this kind of stuff might go into the store... but even if it does we'll probably get it all out via a hook.